modules/cf-core/src/models/state-channel.ts

```diff
@@ -596,6 +596,12 @@ export class StateChannel {
     };
   }
 
+  /**
+   * The state channel JSON object should *always* have an associated proxy
+   * bytecode. There is no case where a JSON version of a state channel is
+   * created that did *not* have an associated bytecode with it
+   *
+   */
```

Comment:
would the plan to make this backwards compatible add a check for byte code and then an insertion here? (and a corresponding assertion in the `toJson` function?

Reply:
No assertion.
Also this comment is out of date, we don't directly care about the proxy bytecode anymore, we care about the factory address (from which we can retrieve the *real* bytecode).
But if no address exists idk, prob try to use the current one from NetworkContext and throw if that one doesn't match